### DISCUSSION TITLE
How does Bromite bypass **intentionally** blocked websites?

### DISCUSSION THREAD

**Discussion by pixincreate** [bromite/bromite#2116] [Q&A]
I use AdAway (root) with many lists that includes Google's Firebase. I tried to access the documentation on @Brave and as usual, AdAway blocked it. I opened the same in Bromite and to my surprise, Bromite bypassed the block that I'd put up to access the website.

Just verify that what I saw was true, I tried opening the same [website](https://firebase.google.com/docs/cloud-messaging/concept-options) on other browsers like Via, Private Browser, Duckduckgo as well, which all blocked the access to me.

I'm curious to know how Bromite bypassed that block, is it a bug (I personally don't think it as a bug but rather censorship circumvention)?

**Comment by csagan5** [comment]
There is no blocking of URLs in Bromite, only subresources are blocked; see https://github.com/bromite/bromite/wiki/AdBlocking

**Reply by pixincreate** [reply]
I think you misunderstood by what I said. Lemme explain it with an example.

`https://0-0.fr`          
This link above was blocked by me personally on AdAway with a script from Energized. Clicking on this link on Bromite opens it directly while the same on other browsers  are blocked (which is what I want) and the page is redirected to 127.0.0.1 with AdAway logo.

I wonder how's that possible. I've legit spent more than a month verifying this. The only other option that is left for me is to clear the app data and check.

It'd be helpful if you or anyone who has AdAway with root access confirm the issue by visiting the link.    
I can share screenshots if required.

**Reply by csagan5** [reply]
> This link above was blocked by me personally on AdAway

AdAway works by hijacking some DNS responses, you probably want to disable Secure DNS in Bromite to not interfere with that.

**Comment by ghost** [comment]


**Reply by pixincreate** [reply]
Nah, private DNS is set up globally but that has nothing to do with what I've mentioned.

I'm facing this issue only with this app and not on other browsers.

In case if the browser has no issue but mine (just because I've made a ton of changes in the flags by means of hardening), I might have to reset the app data which is quite not easy for me. 

And that's why if anyone with AdAway and root can confirm this, it'd be a lot more helpful.

**Reply by uazo** [reply]
> Are you using DoH?
> has nothing to do with what I've mentioned.

why? if you have set the doh in bromite then the request is not passed to adway.

> And that's why if anyone with AdAway and root can confirm this

me, but having active the doh the requests do not arrive to adawys, I confirm it, you can try it with the log integrated in adway

**Reply by pixincreate** [reply]
> why?

Private DNS in device settings is set to Quad9. 

>  if you have set the doh in bromite then the request is not passed to adway.

Oh, wait! I think this solved the issue that I was facing for months. I'd been into Quad9 DNS for years now and is set-up in Private DNS in device settings. I checked the DNS in Bromite and was set to Quad9 under custom DNS. i just changed that to Automatic (System DNS which is again Quad9) and AdAway blocked the website!

Thank you!

Edit: Mr. Uazo, kindly post the same answer separately and I'll mark it as the answer as it resolved my issue. :)
